### PR TITLE
TypeScript Tooling

### PR DESCRIPTION
**Pull request by lemueldls**
### Refactoring to TypeScript

All tools including `api`, `bot`, and `map` have been refactored and converted into TypeScript.

Although the map data isn't entirely "typed", the old code has changed into using classes
and better import solutions. The Node.js `scripts` use `ts-node` to run each tools script.
"Typing" the entirety of tools' map data can be a future suggestion to PR.

### Moving to `package` directory

Instead of having the `tools` at root, I've moved them in with the rest of the packages.

Node.js `scripts` haved been changed to `lerna run` commands (`helper` and `map`) to link to `tools`.
The reason being both for consistency of having project folders inside the `package` directory,
and the other being conveniences of importing by project alias as follows.

```typescript
import worldClient from '@kaetram/server/data/map/world_client.json';
import worldServer from '@kaetram/server/data/map/world_server.json';
```

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -22,16 +22,16 @@
         "no-fallthrough": "off",
         "prettier/prettier": "warn",
         "indent": [
-            "error",
+            "warn",
```
Comment (frycz) [inline .eslintrc:25]: Why are we changing error levels?

Comment (lemueldls) [reply .eslintrc:25]: I just believed small **stylistic** linting rules aren't worth something like failing a CI test, or stops a build entirely if linting were checked.

We could also simply revert this and leave rules more strict for the future.

Comment (frycz) [reply .eslintrc:25]: I wouldn't change the levels. Failing a CI test and stopping a build is a way to enforce the rules and to be 100% sure that the code is of high quality.

If we think that some rules are less important than the other ones, I would just remove them.

File: packages/tools/bot/bot.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                else this.handlePackets(connection, JSON.parse(message).shift());
+            } else this.handlePackets(connection, message, 'utf8');
+        });
+
+        connection.on('disconnect', () => {
+            //
+        });
+    }
+
+    handlePackets(connection: SocketIOClient.Socket, message: never[], type?: string): void {
```
Comment (frycz) [inline packages/tools/bot/bot.ts:94]: Why `message` is an array of `never`? It means that the array will always be empty so there is no need for this argument.

File: packages/tools/map/exportmap.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                if (file.startsWith('tilesheet-'))
+                    fs.copyFileSync(`${source}/${file}`, `${destination}/${file}`);
+            });
+
+            console.log(`Finished copying tilesets to ${relative(destination)}/`);
+        });
+    }
+}
+
+// String.prototype.format = function () {
```
Comment (frycz) [inline packages/tools/map/exportmap.ts:116]: Maybe we should remove commented code

**Review by frycz — APPROVED** [review]

**Pull request merged**